### PR TITLE
fix broken links and replace latest docs links

### PR DESCRIPTION
**Pull request by ghost**
Related Issues: #34488 #34491

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -317,7 +317,7 @@ runtest harness).
   - **Homepage:** <https://julialang.org>
   - **Community:** <https://julialang.org/community/>
   - **Source code:** <https://github.com/JuliaLang/julia>
-  - **Documentation:** <https://docs.julialang.org/>
+  - **Documentation:** <https://docs.julialang.org>
```
Comment (fredrikekre) [inline CONTRIBUTING.md]: Why this change (and the similar ones throughout the PR)?

File: doc/man/julia.1 [hunk 2]
```diff
@@ -47,7 +47,7 @@ by defining functions and overloading them for different combinations
 of argument types, which can also be user-defined.
 For a more in-depth discussion of the rationale and advantages of Julia
 over other systems, please see the online manual:
-https://docs.julialang.org/en/latest/
+https://docs.julialang.org
```
Comment (fredrikekre) [inline doc/man/julia.1]:
```suggestion
https://docs.julialang.org/
```

File: HISTORY.md [hunk 3]
```diff
@@ -1167,7 +1167,7 @@ This section lists changes that do not have deprecation warnings.
     to evaluate the wrapper. Consequently, package authors generally need to specialize
     `copy` and `copyto!` methods rather than `broadcast` and `broadcast!`. This also allows
     for more customization and control of fused broadcasts. See the
-    [Interfaces chapter](https://docs.julialang.org/en/latest/manual/interfaces/#man-interfaces-broadcasting-1)
+    [Interfaces chapter](https://docs.julialang.org/en/v1/manual/interfaces/#man-interfaces-broadcasting-1)
```
Comment (fredrikekre) [inline HISTORY.md]: I'm not sure if it makes sense to update the HISTORY file. Did you check that the updated links still make sense for the context?

Comment (ViralBShah) [reply HISTORY.md:1170]: I think it would be ok. v1 would certainly be better than latest in HISTORY, I think.

**Comment by ViralBShah** [comment]
I am ok merging this one. On the whole, removing `latest` is a good idea.

**Comment by ViralBShah** [comment]
Can we rebase this PR?
